File: docs/DSA/neetcode150/src/py/longest_substring_without_repeating_characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        idxMap = dict()
        l = 0
        for r, ch in enumerate(s):
            if ch in idxMap and idxMap[ch] >= l:
                l = idxMap[ch]+1
            idxMap[ch] = r
        return len(s)-l

File: docs/DSA/src/go/longest_substring_without_repeating_characters.go
package main

func lengthOfLongestSubstring(s string) int {
	idxMap := make(map[rune]int)
	var result, l int
	for r, ch := range s {
		if val, ok := idxMap[ch]; ok && val >= l {
			l = val + 1
		}
		idxMap[ch] = r
		result = max(result, r-l+1)
	}
	return result
}
